tower/src/util/boxed/layer.rs

```diff
@@ -0,0 +1,86 @@
+use crate::util::BoxService;
+use std::{fmt, sync::Arc};
+use tower_layer::{layer_fn, Layer};
+use tower_service::Service;
+
+/// A boxed `Layer` trait object.
+///
+/// [`BoxLayer`] turns a layer into a trait object, allowing the output service to be dynamic.
```

Comment:
not sure if `BoxLayer` needs to be a link, because this _is_ the docs for `BoxLayer`...I think this should be
```suggestion
/// `BoxLayer` turns a [`Layer`] into a trait object, allowing the output [`Service`] to be dynamic.
```

Reply:
can we maybe explain what "allowing the output service to be dynamic" actually means a little more clearly? This boxes the `Service` as well as the `Layer`, so that the layer can emit more than one kind of `Service`. Maybe something like
```suggestion
/// [`BoxLayer`] turns a layer into a trait object, allowing both the [`Layer`] itself
/// and the output [`Service`] to be dynamic, while having consistent types.
///
/// This [`Layer`] produces [`BoxService`] instances erasing the type of the 
/// [`Service`] produced by the wrapped [`Layer`].
```